src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -20,11 +20,17 @@ class MessageDispatcher : IMessageDispatcher
 
         readonly MessageSenderRegistry messageSenderRegistry;
         readonly string topicName;
+        readonly NativeMessageCustomizerCallback? customizerCallback;
 
-        public MessageDispatcher(MessageSenderRegistry messageSenderRegistry, string topicName)
+        public MessageDispatcher(
+            MessageSenderRegistry messageSenderRegistry,
+            string topicName,
+            NativeMessageCustomizerCallback? customizerCallback = null
+        )
         {
             this.messageSenderRegistry = messageSenderRegistry;
             this.topicName = topicName;
+            this.customizerCallback = customizerCallback;
```

Comment:
By assigning a static empty version when null you remove the null check on the hot path

Reply:
Applied